pkg/validator/validator.go

```diff
@@ -115,7 +122,32 @@ func IsValidSubdotName(str string) error {
 	return nil
 }
 
+func IsValidSnapshotName(str string) error {
+	if str == "" {
+		return ErrEmptySnapshot
+	}
+
+	if !rxSubdot.MatchString(str) {
+		return ErrInvalidSnapshotName
+	}
+
+	return nil
+}
+
 // ReplaceUUID replace UUID in string
 func ReplaceUUID(str, replace string) string {
 	return rxUUIDPattern.ReplaceAllString(str, replace)
 }
+
+type ValidatorFunc func(string) error
+
+// Return false (and respond with HTTP error) if the value doesn't validate.
+func EnsureValid(value string, validator ValidatorFunc, resp http.ResponseWriter) bool {
```

Comment:
Small suggestion - could this function be called something that indicates it handles the HTTP response also?

`EnsureValid` feels like it is just checking the validity of the value - naming things is hard though and the best I can come up with is `EnsureValidWithResponse` - not a biggie :-)

Reply:
Good point, done.